File: server/src/Makc2021.Layer1/Query/QueryHelper.cs
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using System;

namespace Makc2021.Layer1.Query
{
    /// <summary>
    /// Помощник запроса.
    /// </summary>
    public class QueryHelper
    {
        #region Public methods

        /// <summary>
        /// Создать код запроса.
        /// </summary>
        /// <returns>Код запроса.</returns>
        public static string CreateQueryCode()
        {
            return Guid.NewGuid().ToString("N").ToUpper();
        }

        #endregion Public methods
    }
}

File: client/makc2021-react-redux-ts/src/Layer1/Query/QueryHelper.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

import { v4 as uuidv4 } from 'uuid';

/**
 * Помощник запроса.
 */
export class QueryHelper {
  /**
   * Создать код запроса.
   * @returns Код запроса.
   */
  static createQueryCode() {
    return uuidv4().replace(/-/, '').toUpperCase(); //uuidv4().replaceAll('-', '').toUpperCase();
  }
}
